autoload/venu/native.vim

```diff
@@ -0,0 +1,107 @@
+function! s:parseMenuLine(line)
+  let l:match = matchlist(a:line, "\\v^( *)([^ ]+)\\s+(.{-})\\s*\\r?$")
+  if len(l:match) == 0
+    return []
+  endif
+  return [len(l:match[1]), l:match[2], l:match[3]]
+endfunction
+
+" Extract <TAB>=^I separated description from name if there exists one
+function! s:extractDesc(name)
+  let l:tmp = matchlist(a:name, '\v^(.{-})\^I(.{-})$')
+  if len(l:tmp) == 0
+    let l:name = a:name
+    let l:desc = ''
+  else
+    let l:name = l:tmp[1]
+    let l:desc = l:tmp[2]
+  endif
+  return [l:name, l:desc]
+endfunction
+
+" Extract &Shortcut and replace escaped ampersands (&&).
+function! s:extractShortcut(name)
+  " These are not some curse words, just some regex to match first unescaped &.
+  let l:tmp = matchend(a:name, "\\v(\\&)@<!%(\\&\\&)*(\\&)(\\&)@!")
+  if l:tmp < 0
+    let l:shortcut = ''
+    let l:name = a:name
+  else
+    let l:shortcut = a:name[l:tmp]
+    let l:name = substitute(
+    \   strpart(a:name, 0, l:tmp - 1).strpart(a:name, l:tmp),
+    \   "&&", "&", "g"
+    \ )
+  endif
+  return [l:name, l:shortcut]
+endfunction
+
+function! venu#native#list()
+  let l:menu_str = execute("menu")
+  let l:root = []
+  let l:prevdepth = 0
+
+  for l:line in split(l:menu_str, "\\n")
+    let l:line = s:parseMenuLine(l:line)
+    if len(l:line) > 0
+      if match(l:line[1], "\\v^\\d+$") == 0
+        let l:indent   = l:line[0]
+        let l:depth    = l:indent / 2
+        let l:priority = str2nr(l:line[1])
+
+        let l:tmp  = s:extractDesc(l:line[2])
+        let l:name = l:tmp[0]
+        let l:desc = l:tmp[1]
+
+        let l:tmp      = s:extractShortcut(l:line[2])
+        let l:name     = l:tmp[0]
+        let l:shortcut = l:tmp[1]
+
+        let l:menu = {
+              \  'name': l:name,
+              \  'desc': l:desc,
+              \  'priority': l:priority,
+              \  'shortcut': l:shortcut,
+              \  'submenus': [],
+              \  'mode': {},
+              \}
+
+        if l:depth == 0
+          call add(l:root, l:menu)
+          let l:stack = []
+        elseif l:depth > l:prevdepth
+          call assert_true(l:depth - l:prevdepth == 1)
+          call add(l:stack[-1].submenus, l:menu)
+        elseif l:depth == l:prevdepth
+          call remove(l:stack, -1)
+          call add(l:stack[-1].submenus, l:menu)
+        else " l:depth < l:prevdepth
+          call remove(l:stack, -(l:prevdepth - l:depth + 1), -1)
+          call add(l:stack[-1].submenus, l:menu)
+        endif
+
+        call add(l:stack, l:menu)
+        let l:prevdepth = l:depth
+
+      else
+      endif
+    endif
+  endfor
+  return l:root
+endfunction
+
+function! s:debugTraverse(l, d)
```

Comment:
More descriptive argument names might be nice. E.g. `list` and `indentation` (though, I'm not sure what the use of indentation is).

Reply:
I didn't care since I was planning to remove this function before merge, but on second thought it might be better if it's kept around. I will rename its args.